src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -28,9 +30,31 @@ open import Relation.Unary using (Pred)
 open import Relation.Binary
 open import Relation.Binary.PropositionalEquality as ≡
   using (_≡_ ; refl ; cong; cong₂; _≢_; inspect)
+open import Relation.Nullary
 
 open PermutationReasoning
 
+
+------------------------------------------------------------------------
+-- Permutations of empty and singleton lists
+
+module _ {a} {A : Set a} where
+
+  ↭-[] : ∀ {xs : List A} → xs ↭ [] → xs ≡ []
+  ↭-[] refl = refl
```

Comment:
I would call this an inversion principle (in the sense of [Coq](https://www-verimag.imag.fr/~monin/Publis/Docs/coq2.pdf)).
I am surprised we do not have more of these: I could only find [`remove-inv`](http://agda.github.io/agda-stdlib/Data.List.Fresh.Membership.Setoid.Properties.html#2206).

Reply:
Concretely what name would you give it then? `↭-[]-inv` or something smaller?

Reply:
`↭-[]-inv` looks good to me.